### PR TITLE
Update to Zig 0.16

### PR DESCRIPTION
**Pull request by BratishkaErik**
Hello, I tried to update this library to Zig 0.16. Unfortunately, there are some leaks in test mode, I don't know yet if they are related to changes in code (I tried to minimize tho), in test runner (copied from https://gist.githubusercontent.com/karlseguin/c6bea5b35e4e8d26af6f81c22cb5d76b/raw/8e82642093d2f4341fb870d18d1fe5b213b2fd11), or something else.

But it works when I just use and import it, so it's already usable (at least for me).

```bash
$ zig-bin-0.16.0 build test
ztl.test_0 (0.01ms)
Value: isTrue (0.03ms)
Value: format (1.00ms)
Value: equal (0.32ms)
writeStringEscaped (0.08ms)
Template: simple (2.14ms)
Template: edge cases (3.39ms)
Template: output literals (1.71ms)
Template: output variable (0.65ms)
Template: space stripping (1.48ms)
Template: escaping (1.37ms)
Template: local and global (0.40ms)
Template: for loop (0.46ms)
Template: error (1.37ms)
Template: runtime error (0.49ms)
Template: map global (0.71ms)
Template: semicolon (1.08ms)
Template: global in function (0.52ms)
Template: string concatenation (2.71ms)
Template: @include (2.62ms)
Template: @include error (0.52ms)
Template: multiple index get (0.80ms)
Template: large (14.56ms)
Template: nested function (0.41ms)
tests:afterEach (0.00ms)
Compiler: local limit (0.46ms)
error(DebugAllocator): memory address 0x74d7a53c0000 leaked:
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:142:26: 0x1097aa2 in rawAlloc (std.zig)
    return a.vtable.alloc(a.ptr, len, alignment, ret_addr);
                         ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:286:40: 0x115f9a5 in allocWithSizeAndAlignment__anon_20539 (std.zig)
    return self.allocBytesWithAlignment(alignment, byte_count, return_address);
                                       ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:274:89: 0x136a24c in allocAdvancedWithRetAddr (std.zig)
    const ptr: [*]align(a.toByteUnits()) T = @ptrCast(try self.allocWithSizeAndAlignment(@sizeOf(T), a, n, return_address));
                                                                                        ^
/opt/zig-bin-0.16.0/lib/std/array_list.zig:1235:56: 0x1369c07 in ensureTotalCapacityPrecise (std.zig)
                const new_memory = try gpa.alignedAlloc(T, alignment, new_capacity);
                                                       ^
/opt/zig-bin-0.16.0/lib/std/array_list.zig:1211:51: 0x1369919 in ensureTotalCapacity (std.zig)
            return self.ensureTotalCapacityPrecise(gpa, growCapacity(new_capacity));
                                                  ^
/opt/zig-bin-0.16.0/lib/std/array_list.zig:1265:41: 0x13693dd in addOne (std.zig)
            try self.ensureTotalCapacity(gpa, newlen);
                                        ^
/opt/zig-bin-0.16.0/lib/std/array_list.zig:904:49: 0x1368fbd in append (std.zig)
            const new_item_ptr = try self.addOne(gpa);
                                                ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:1446:35: 0x1434d73 in beginScope (ztl.zig)
            try self.scopes.append(self.arena, .{
                                  ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:177:32: 0x14351be in compile (ztl.zig)
            try self.beginScope(false);
                               ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:3674:18: 0x1861984 in testReturnValueWithApp__anon_101104 (ztl.zig)
        c.compile("<% " ++ src ++ "%>") catch |err| {
                 ^
(additional stack frames may have been skipped...)

error(DebugAllocator): memory address 0x74d7a53a0000 leaked:
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:142:26: 0x10a4ff2 in rawAlloc (std.zig)
    return a.vtable.alloc(a.ptr, len, alignment, ret_addr);
                         ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:286:40: 0x10a4e75 in allocWithSizeAndAlignment__anon_9388 (std.zig)
    return self.allocBytesWithAlignment(alignment, byte_count, return_address);
                                       ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:274:89: 0x10a4b0c in allocAdvancedWithRetAddr (std.zig)
    const ptr: [*]align(a.toByteUnits()) T = @ptrCast(try self.allocWithSizeAndAlignment(@sizeOf(T), a, n, return_address));
                                                                                        ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:299:44: 0x138ed7c in toBytes (ztl.zig)
            const buf = try allocator.alloc(u8, 9 + code.pos + script.pos + data.pos);
                                           ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:3678:40: 0x186256d in testReturnValueWithApp__anon_101119 (ztl.zig)
        break :blk try c.writer.toBytes(allocator);
                                       ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:3653:31: 0x18638b7 in testReturnValue__anon_101100 (ztl.zig)
    try testReturnValueWithApp(struct {
                              ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:2015:24: 0x1889e05 in test.Compiler: arithmetic (ztl.zig)
    try testReturnValue(.{ .i64 = 9 }, "return 1 + 8;");
                       ^
/tmp/tmp.kstEoQdTP6/ztl/test_runner.zig:97:30: 0x122a828 in main (test_runner.zig)
        const result = t.func();
                             ^
/opt/zig-bin-0.16.0/lib/std/start.zig:737:30: 0x122c09e in callMain (std.zig)
    return wrapMain(root.main(.{
                             ^
/opt/zig-bin-0.16.0/lib/std/start.zig:190:5: 0x1229b81 in _start (std.zig)
    asm volatile (switch (native_arch) {
    ^
(additional stack frames may have been skipped...)

error(DebugAllocator): memory address 0x74d7a5380000 leaked:
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:142:26: 0x10a4ff2 in rawAlloc (std.zig)
    return a.vtable.alloc(a.ptr, len, alignment, ret_addr);
                         ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:286:40: 0x10a4e75 in allocWithSizeAndAlignment__anon_9388 (std.zig)
    return self.allocBytesWithAlignment(alignment, byte_count, return_address);
                                       ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:274:89: 0x10a4b0c in allocAdvancedWithRetAddr (std.zig)
    const ptr: [*]align(a.toByteUnits()) T = @ptrCast(try self.allocWithSizeAndAlignment(@sizeOf(T), a, n, return_address));
                                                                                        ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:299:44: 0x13ad07c in toBytes (ztl.zig)
            const buf = try allocator.alloc(u8, 9 + code.pos + script.pos + data.pos);
                                           ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:3678:40: 0x186522d in testReturnValueWithApp__anon_101165 (ztl.zig)
        break :blk try c.writer.toBytes(allocator);
                                       ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:3660:31: 0x1865b1b in testReturnValue__anon_101139 (ztl.zig)
    try testReturnValueWithApp(struct {
                              ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:2016:24: 0x1889e6f in test.Compiler: arithmetic (ztl.zig)
    try testReturnValue(.{ .i64 = -1 }, "return 10 - 11;");
                       ^
/tmp/tmp.kstEoQdTP6/ztl/test_runner.zig:97:30: 0x122a828 in main (test_runner.zig)
        const result = t.func();
                             ^
/opt/zig-bin-0.16.0/lib/std/start.zig:737:30: 0x122c09e in callMain (std.zig)
    return wrapMain(root.main(.{
                             ^
/opt/zig-bin-0.16.0/lib/std/start.zig:190:5: 0x1229b81 in _start (std.zig)
    asm volatile (switch (native_arch) {
    ^
(additional stack frames may have been skipped...)

error(DebugAllocator): memory address 0x74d7a5360000 leaked:
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:142:26: 0x1097aa2 in rawAlloc (std.zig)
    return a.vtable.alloc(a.ptr, len, alignment, ret_addr);
                         ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:286:40: 0x115f9a5 in allocWithSizeAndAlignment__anon_20539 (std.zig)
    return self.allocBytesWithAlignment(alignment, byte_count, return_address);
                                       ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:274:89: 0x136a24c in allocAdvancedWithRetAddr (std.zig)
    const ptr: [*]align(a.toByteUnits()) T = @ptrCast(try self.allocWithSizeAndAlignment(@sizeOf(T), a, n, return_address));
                                                                                        ^
/opt/zig-bin-0.16.0/lib/std/array_list.zig:1235:56: 0x1369c07 in ensureTotalCapacityPrecise (std.zig)
                const new_memory = try gpa.alignedAlloc(T, alignment, new_capacity);
                                                       ^
/opt/zig-bin-0.16.0/lib/std/array_list.zig:1211:51: 0x1369919 in ensureTotalCapacity (std.zig)
            return self.ensureTotalCapacityPrecise(gpa, growCapacity(new_capacity));
                                                  ^
/opt/zig-bin-0.16.0/lib/std/array_list.zig:1265:41: 0x13693dd in addOne (std.zig)
            try self.ensureTotalCapacity(gpa, newlen);
                                        ^
/opt/zig-bin-0.16.0/lib/std/array_list.zig:904:49: 0x1368fbd in append (std.zig)
            const new_item_ptr = try self.addOne(gpa);
                                                ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:1446:35: 0x13dff43 in beginScope (ztl.zig)
            try self.scopes.append(self.arena, .{
                                  ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:177:32: 0x13e038e in compile (ztl.zig)
            try self.beginScope(false);
                               ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:3674:18: 0x186b704 in testReturnValueWithApp__anon_101279 (ztl.zig)
        c.compile("<% " ++ src ++ "%>") catch |err| {
                 ^
(additional stack frames may have been skipped...)

error(DebugAllocator): memory address 0x74d7a5340000 leaked:
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:142:26: 0x1097aa2 in rawAlloc (std.zig)
    return a.vtable.alloc(a.ptr, len, alignment, ret_addr);
                         ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:286:40: 0x115f9a5 in allocWithSizeAndAlignment__anon_20539 (std.zig)
    return self.allocBytesWithAlignment(alignment, byte_count, return_address);
                                       ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:274:89: 0x136a24c in allocAdvancedWithRetAddr (std.zig)
    const ptr: [*]align(a.toByteUnits()) T = @ptrCast(try self.allocWithSizeAndAlignment(@sizeOf(T), a, n, return_address));
                                                                                        ^
/opt/zig-bin-0.16.0/lib/std/array_list.zig:1235:56: 0x1369c07 in ensureTotalCapacityPrecise (std.zig)
                const new_memory = try gpa.alignedAlloc(T, alignment, new_capacity);
                                                       ^
/opt/zig-bin-0.16.0/lib/std/array_list.zig:1211:51: 0x1369919 in ensureTotalCapacity (std.zig)
            return self.ensureTotalCapacityPrecise(gpa, growCapacity(new_capacity));
                                                  ^
/opt/zig-bin-0.16.0/lib/std/array_list.zig:1265:41: 0x13693dd in addOne (std.zig)
            try self.ensureTotalCapacity(gpa, newlen);
                                        ^
/opt/zig-bin-0.16.0/lib/std/array_list.zig:904:49: 0x1368fbd in append (std.zig)
            const new_item_ptr = try self.addOne(gpa);
                                                ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:1446:35: 0x140a903 in beginScope (ztl.zig)
            try self.scopes.append(self.arena, .{
                                  ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:177:32: 0x140ad4e in compile (ztl.zig)
            try self.beginScope(false);
                               ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:3674:18: 0x1877844 in testReturnValueWithApp__anon_101508 (ztl.zig)
        c.compile("<% " ++ src ++ "%>") catch |err| {
                 ^
(additional stack frames may have been skipped...)

error(DebugAllocator): memory address 0x74d7a5320000 leaked:
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:142:26: 0x10a4ff2 in rawAlloc (std.zig)
    return a.vtable.alloc(a.ptr, len, alignment, ret_addr);
                         ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:286:40: 0x10a4e75 in allocWithSizeAndAlignment__anon_9388 (std.zig)
    return self.allocBytesWithAlignment(alignment, byte_count, return_address);
                                       ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:274:89: 0x10a4b0c in allocAdvancedWithRetAddr (std.zig)
    const ptr: [*]align(a.toByteUnits()) T = @ptrCast(try self.allocWithSizeAndAlignment(@sizeOf(T), a, n, return_address));
                                                                                        ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:331:46: 0x13011a9 in write (ztl.zig)
                    buf = try allocator.alloc(u8, initial_size);
                                             ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:60:36: 0x140ab75 in op (ztl.zig)
            return self.frame.write(self.allocator, &.{@intFromEnum(op_code)});
                                   ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:68:24: 0x1411b44 in opWithData (ztl.zig)
            try self.op(op_code);
                       ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:216:35: 0x142a365 in f64 (ztl.zig)
            return self.opWithData(.CONSTANT_F64, std.mem.asBytes(&value));
                                  ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:993:54: 0x142a4de in number (ztl.zig)
                .FLOAT => |value| try self.writer.f64(value),
                                                     ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:935:31: 0x1421fa6 in parsePrecedence (ztl.zig)
                    try prefix(self, can_assign);
                              ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:907:40: 0x1421a56 in expression (ztl.zig)
            return self.parsePrecedence(.ASSIGNMENT);
                                       ^
(additional stack frames may have been skipped...)


================================================================================
"Compiler: arithmetic" - Memory Leak
================================================================================
Compiler: arithmetic (1.19ms)
Compiler: not (0.37ms)
error(DebugAllocator): memory address 0x74d7a366a000 leaked:
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:142:26: 0x10a4ff2 in rawAlloc (std.zig)
    return a.vtable.alloc(a.ptr, len, alignment, ret_addr);
                         ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:286:40: 0x10a4e75 in allocWithSizeAndAlignment__anon_9388 (std.zig)
    return self.allocBytesWithAlignment(alignment, byte_count, return_address);
                                       ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:274:89: 0x10a4b0c in allocAdvancedWithRetAddr (std.zig)
    const ptr: [*]align(a.toByteUnits()) T = @ptrCast(try self.allocWithSizeAndAlignment(@sizeOf(T), a, n, return_address));
                                                                                        ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:331:46: 0x13011a9 in write (ztl.zig)
                    buf = try allocator.alloc(u8, initial_size);
                                             ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:60:36: 0x13e01b5 in op (ztl.zig)
            return self.frame.write(self.allocator, &.{@intFromEnum(op_code)});
                                   ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:68:24: 0x13e7184 in opWithData (ztl.zig)
            try self.op(op_code);
                       ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:212:35: 0x13f8d33 in i64 (ztl.zig)
            return self.opWithData(.CONSTANT_I64, std.mem.asBytes(&value));
                                  ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:992:56: 0x13fffd8 in number (ztl.zig)
                .INTEGER => |value| try self.writer.i64(value),
                                                       ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:935:31: 0x13f7b06 in parsePrecedence (ztl.zig)
                    try prefix(self, can_assign);
                              ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:907:40: 0x13f75b6 in expression (ztl.zig)
            return self.parsePrecedence(.ASSIGNMENT);
                                       ^
(additional stack frames may have been skipped...)


================================================================================
"Compiler: comparison int" - Memory Leak
================================================================================
Compiler: comparison int (0.55ms)
================================================================================
panic running "Compiler: comparison float"
================================================================================
thread 301 panic: Invalid free
/tmp/tmp.kstEoQdTP6/ztl/test_runner.zig:290:31: 0x107b9a9 in panicFn (test_runner.zig)
        std.debug.defaultPanic(msg, first_trace_addr);
                              ^
/opt/zig-bin-0.16.0/lib/std/heap/debug_allocator.zig:553:21: 0x136d70c in resizeLarge (std.zig)
                    @panic("Invalid free");
                    ^
/opt/zig-bin-0.16.0/lib/std/heap/debug_allocator.zig:839:40: 0x136eea2 in resize (std.zig)
                return self.resizeLarge(memory, alignment, new_len, return_address, false) != null;
                                       ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:148:27: 0x10992e3 in rawResize (std.zig)
    return a.vtable.resize(a.ptr, memory, alignment, new_len, ret_addr);
                          ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:142:26: 0x10a4ff2 in rawAlloc (std.zig)
    return a.vtable.alloc(a.ptr, len, alignment, ret_addr);
                         ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:286:40: 0x10a4e75 in allocWithSizeAndAlignment__anon_9388 (std.zig)
    return self.allocBytesWithAlignment(alignment, byte_count, return_address);
                                       ^
/opt/zig-bin-0.16.0/lib/std/mem/Allocator.zig:274:89: 0x10a4b0c in allocAdvancedWithRetAddr (std.zig)
    const ptr: [*]align(a.toByteUnits()) T = @ptrCast(try self.allocWithSizeAndAlignment(@sizeOf(T), a, n, return_address));
                                                                                        ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:331:46: 0x13011a9 in write (ztl.zig)
                    buf = try allocator.alloc(u8, initial_size);
                                             ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:60:36: 0x1434fe5 in op (ztl.zig)
            return self.frame.write(self.allocator, &.{@intFromEnum(op_code)});
                                   ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:68:24: 0x1439774 in opWithData (ztl.zig)
            try self.op(op_code);
                       ^
/tmp/tmp.kstEoQdTP6/ztl/src/byte_code.zig:216:35: 0x1455b95 in f64 (ztl.zig)
            return self.opWithData(.CONSTANT_F64, std.mem.asBytes(&value));
                                  ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:993:54: 0x1455d0e in number (ztl.zig)
                .FLOAT => |value| try self.writer.f64(value),
                                                     ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:935:31: 0x144d216 in parsePrecedence (ztl.zig)
                    try prefix(self, can_assign);
                              ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:907:40: 0x144ccc6 in expression (ztl.zig)
            return self.parsePrecedence(.ASSIGNMENT);
                                       ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:732:44: 0x144449e in statement (ztl.zig)
                        try self.expression();
                                           ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:375:50: 0x143bd02 in declaration (ztl.zig)
                    else => return self.statement(),
                                                 ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:180:37: 0x143571c in compile (ztl.zig)
                try self.declaration();
                                    ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:3674:18: 0x1820a74 in testReturnValueWithApp__anon_99961 (ztl.zig)
        c.compile("<% " ++ src ++ "%>") catch |err| {
                 ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:3646:31: 0x1822943 in testReturnValue__anon_99957 (ztl.zig)
    try testReturnValueWithApp(struct {
                              ^
/tmp/tmp.kstEoQdTP6/ztl/src/compiler.zig:2072:24: 0x183a29d in test.Compiler: comparison float (ztl.zig)
    try testReturnValue(.{ .bool = false }, "return 1.13 > 1.13;");
                       ^
/tmp/tmp.kstEoQdTP6/ztl/test_runner.zig:97:30: 0x122a828 in main (test_runner.zig)
        const result = t.func();
                             ^
/opt/zig-bin-0.16.0/lib/std/start.zig:737:30: 0x122c09e in callMain (std.zig)
    return wrapMain(root.main(.{
                             ^
/opt/zig-bin-0.16.0/lib/std/start.zig:190:5: 0x1229b81 in _start (std.zig)
    asm volatile (switch (native_arch) {
    ^
test
└─ run test failure
error: process terminated with signal ABRT
failed command: ./.zig-cache/o/926a6b06c608ea40895f98418b6815c4/test

Build Summary: 1/3 steps succeeded (1 failed)
test transitive failure
└─ run test failure

error: the following build command failed with exit code 1:
.zig-cache/o/4ce4b4aba0d87bdf13b305a078630cfb/build /opt/zig-bin-0.16.0/zig /opt/zig-bin-0.16.0/lib /tmp/tmp.kstEoQdTP6/ztl .zig-cache /home/bratishkaerik/.cache/zig --seed 0xe844b652 -Z2b45f9fc27a1cad9 test
```

### CODE REVIEW

File: src/t.zig [hunk 1]
```diff
@@ -19,7 +19,7 @@ pub fn reset() void {
 
 pub fn createListRef(values: []Value) Value {
     const ref = arena.allocator().create(Value.Ref) catch unreachable;
-    ref.* = .{ .value = .{ .list = .{ .items = values } } };
+    ref.* = .{ .value = .{ .list = .{ .items = values, .capacity = 0 } } };
```
Comment (BratishkaErik) [inline src/t.zig]: `capacity = 0` is strange, but it matches old default value.

File: src/value.zig [hunk 2]
```diff
@@ -531,34 +531,34 @@ test "Value: equal" {
...
+    //     aw.clearRetainingCapacity();
     // }
 
     // {
-    //     try writeStringEscaped(buf.writer(), "<>\"'&");
-    //     try t.expectString("&lt;&gt;&#34;&#39;&amp;", buf.items);
-    //     buf.clearRetainingCapacity();
+    //     try writeStringEscaped(&aw.writer, "<>\"'&");
+    //     try t.expectString("&lt;&gt;&#34;&#39;&amp;", aw.written());
+    //     aw.clearRetainingCapacity();
```
Comment (BratishkaErik) [inline src/value.zig]: I updated commented-out code only here, other commented places were not touched

File: src/vm.zig [hunk 3]
```diff
@@ -587,9 +588,15 @@ pub fn VM(comptime App: type) type {
...
-                    var buffer = Value.Buffer{};
+                    var buffer: Value.Buffer = .empty;
                     try buffer.appendSlice(self._allocator, l);
-                    try right.write(buffer.writer(self._allocator), false);
+                    {
+                        var aw: std.Io.Writer.Allocating = .fromArrayList(self._allocator, &buffer);
+                        defer buffer = aw.toArrayList();
+                        right.write(&aw.writer, false) catch |err| switch (err) {
+                            error.WriteFailed => return error.OutOfMemory, // See ArrayList.print
+                        };
```
Comment (BratishkaErik) [inline src/vm.zig:597]: Made this so that return error set is not changed.

File: src/ztl.zig [hunk 4]
```diff
@@ -20,30 +20,19 @@ pub fn Functions(comptime A: type) type {
...
     if (App == void or @hasDecl(App, "ZtlFunctions") == false) {
-        return @Type(.{
-            .@"enum" = .{
-                .decls = &.{},
-                .tag_type = u8,
-                .fields = &.{.{ .name = "", .value = 0 }}, // HACK, std.meta.stringToEnum doesn't work on an empty enum, lol what?
-                .is_exhaustive = true,
-            },
-        });
+        return @Enum(u8, .exhaustive, &.{""}, &.{0}); // HACK, std.meta.stringToEnum doesn't work on an empty enum, lol what?
```
Comment (BratishkaErik) [inline src/ztl.zig:23]: Left comment as-is, didn't check if hack is still needed.

**Comment by karlseguin** [comment]
I made some tweaks to  your branch, and fixed a few things. Looks good to me. Do you want to finalize anything?

**Comment by BratishkaErik** [comment]
> I made some tweaks to your branch, and fixed a few things. Looks good to me. Do you want to finalize anything?

Just updated example in README to compile. Feel free to merge when you are ready :)

**Comment by BratishkaErik** [comment]
Thanks!
